sqlstore.go

```diff
@@ -48,23 +54,44 @@ func (f sqlStoreFactory) Create(sessionID SessionID) (msgStore MessageStore, err
 			return nil, err
 		}
 	}
-	return newSQLStore(sessionID, sqlDriver, sqlDataSourceName, sqlConnMaxLifetime)
+	sqlConnMaxIdle := 10
+	if sessionSettings.HasSetting(config.SQLStoreConnMaxIdle) {
+		sqlConnMaxIdle, err = sessionSettings.IntSetting(config.SQLStoreConnMaxIdle)
+		if err != nil {
+			return nil, err
+		}
+	}
+	sqlConnMaxOpen := 20
+	if sessionSettings.HasSetting(config.SQLStoreConnMaxOpen) {
+		sqlConnMaxOpen, err = sessionSettings.IntSetting(config.SQLStoreConnMaxOpen)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return newSQLStore(sessionID, sqlDriver, sqlDataSourceName, dbSettings{
+		connMaxLifetime: sqlConnMaxLifetime,
+		connMaxIdle:     sqlConnMaxIdle,
+		connMaxOpen:     sqlConnMaxOpen,
+	})
 }
 
-func newSQLStore(sessionID SessionID, driver string, dataSourceName string, connMaxLifetime time.Duration) (store *sqlStore, err error) {
+func newSQLStore(sessionID SessionID, driver string, dataSourceName string, dbs dbSettings) (store *sqlStore, err error) {
 	store = &sqlStore{
 		sessionID:          sessionID,
 		cache:              &memoryStore{},
 		sqlDriver:          driver,
 		sqlDataSourceName:  dataSourceName,
-		sqlConnMaxLifetime: connMaxLifetime,
+		sqlConnMaxLifetime: dbs.connMaxLifetime,
```

Comment:
can `sqlConnMaxLifetime` be removed from `sqlStore` - does not look like it's being used anymore?

Reply:
maybe you can keep `dbSettings` in there

Reply:
That implementation is coming from upstream. We are OK with diverging?

Reply:
sure, we're already pretty diverged

Reply:
Done